transit_odp/timetables/dataframes.py

```diff
@@ -302,19 +305,138 @@ def flexible_operation_period_to_dataframe(flexible_vechicle_journeys):
+        "operational": operating_profile["operational"],
+        "day_of_week": operating_profile["day_of_week"],
+        "exceptions_operational": None,
+        "exceptions_date": None,
+    }
+
+    if is_exceptions:
+        operating_profile["exceptions_operational"] = exceptions_operational
+        if date:
+            operating_profile["exceptions_date"] = date
+
+    return operating_profile
+
+
+def get_operating_profiles_for_all_exceptions(
+    operating_profile,
+    operations=None,
+    df_bank_holidays_from_db=pd.DataFrame(),
+    is_bank_holiday_exception=False,
+    is_special_operation=False,
+    is_days_of_operation=False,
+):
+    operating_profile_list = []
+
+    if is_bank_holiday_exception:
+        for holiday in operations.children:
+            date = None
+            no_bank_holidays = False
+            if holiday.tag_localname == "OtherPublicHoliday":
+                date = datetime.strptime(
+                    holiday.get_element(["Date"]).text, "%Y-%m-%d"
+                ).date()
+            else:
+                filtered_df = df_bank_holidays_from_db.loc[
+                    df_bank_holidays_from_db["txc_element"] == holiday.tag_localname,
+                    "date",
+                ]
+
+                if not filtered_df.empty:
+                    date = filtered_df.values[0]
+                else:
+                    no_bank_holidays = True
+            if not no_bank_holidays:
+                operating_profile_list.append(
+                    get_operating_profile_with_exception(
+                        operating_profile,
+                        date,
+                        is_days_of_operation,
+                        is_exceptions=True,
+                    )
+                )
+
+    elif is_special_operation:
+        if operations:
+            date_range = operations.get_elements_or_none(["DateRange"])
+            for range in date_range:
+                start_date = datetime.strptime(
+                    range.get_element(["StartDate"]).text, "%Y-%m-%d"
+                ).date()
+                end_date = datetime.strptime(
+                    range.get_element(["EndDate"]).text, "%Y-%m-%d"
+                ).date()
+
+                while start_date <= end_date:
+                    operating_profile_list.append(
+                        get_operating_profile_with_exception(
+                            operating_profile,
+                            start_date,
+                            is_days_of_operation,
+                            is_exceptions=True,
+                        )
+                    )
+                    start_date = start_date + timedelta(days=1)
+    else:
+        operating_profile_list.append(
+            get_operating_profile_with_exception(operating_profile)
+        )
+
+    return operating_profile_list
+
+
+def populate_operating_profiles(
+    operating_profile_services, vehicle_journey_code, service_ref
```

Comment:
The operating profiles could be from vehicle journeys as well. So can we keep the name generic as operating_profiles?

Reply:
done